react-native-scripts/src/scripts/init.js

```diff
@@ -9,14 +9,14 @@ import log from '../util/log';
 
 // UPDATE DEPENDENCY VERSIONS HERE
 const DEFAULT_DEPENDENCIES = {
-  expo: '^16.0.0',
+  expo: '^17.0.0',
   react: '16.0.0-alpha.6',
-  'react-native': '^0.43.4',
+  'react-native': '^0.44.0',
```

Comment:
The ExpoKit detach flow also needs to have versions updated b/c we replace the vanilla RN version with the tarball URL there.